### PR TITLE
Add GitHub action to update the buf spec file in response to an API commit

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: buf-api-update/src/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    const updatedContents = contents.replace(regex, `generate buf.build/authzed/api:${apiCommit} `);
+    return [updatedContents, contents !== updatedContents]
+  },
+  'gradle': (contents: string, apiCommit: string) => {
+    // Format: `def authzedProtoCommit = "c9dc57b6f25666952f736f5b3ba621397b5e09a3"`
+    const regex = new RegExp(`def authzedProtoCommit = "(.+)"`);
+    if (!regex.test(contents)) {
+      throw new Error(`missing expected generate buf command in shell script`)
```
Comment (samkim) [inline buf-api-update/src/handlers.ts:19]:
```suggestion
            throw new Error(`missing expected var definition in gradle file`);
```

File: buf-api-update/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      return
+    }    
+
+    // Write the updated spec file.
+    const writeFile = util.promisify(fs.writeFile);
+    await writeFile(specFilePath, updatedContents);
+
+    // Done.
+    core.setOutput("updated", 'true')  
+    console.log(`successfully generated updated API protos`)
```
Comment (samkim) [inline buf-api-update/src/index.ts:37]: Is this action intended to run the updated script as well or only update the script file?

Comment (josephschorr) [reply buf-api-update/src/index.ts:37]: Only update it

**Comment by josephschorr** [comment]
Updated

**Review by samkim — APPROVED** [review]
LGTM

**Pull request merged**